sdk/storage/azure-storage-common/inc/azure/storage/common/file_io.hpp

```diff
@@ -4,7 +4,13 @@
 #pragma once
 
 #ifdef _WIN32
-#include <Windows.h>
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#ifndef NOMINMAX
+#define NOMINMAX
```

Comment:
here's a thought, macros are not encouraged to be used according to our guidelines. We have `std::min` and `std::max` in `<algorithm>` which is always better than `min` and `max` macros in `Windows.h`.

We have `NOMINMAX` defined in CMakeFile, so this define here is unnecessary

Reply:
then, okay.

Reply:
I very much dislike MIN and MAX macros that windows.h defines too, and they are harmful - they break the build, break the valid C++ code, and only present on Windows. But I am not removing them, I am just moving the definition from CMakeFile down to the code, motivation being the one I put in my other comment replying to your question about the motivation other comment.